content/blog/survival-analysis-parsnip-adjacent/index.Rmd

```diff
@@ -0,0 +1,227 @@
+
+```{r predict}
+f_pred <- predict(f_fit, new_data = bladder_test,
+                  type = "survival", time = c(5, 10, 15, 20))
+
+# 3 rows for 3 observations
+f_pred 
+# 4 rows for 4 time points
+f_pred$.pred[[1]]
+```
+
+Usually, the linear predictor is the linear combination of the predictors, weighted with the coefficients, often denoted as $x^{T} \beta$. This is what is being returned by the `glmnet` engine for `proportional_hazards()` models and by the `mboost` engine for `boost_tree()` models (e.g., with mode `"censored regression"`). For proportional hazards models, the linear predictor describes _relative_ risk, i.e., it refers to a pair of observations: the observation $x$ and a reference observation with value 0. The survival package by default centers the observation with the mean to allow for an interpretation of risk relative to a "standard" observation. For consistency with other engines, prediction of type `linear_pred` from `proportional_hazards()` models with engine `survival` also do not center the predictors.
+
+The prodlim package let's us simulate survival data with two covariates `X1` (binary) and `X2` (numeric) which both have an effect of $exp(1)$ on the hazard of the unobserved event time.
+
+```{r, simulate}
+library(tidymodels)
+library(ggplot2)
+library(prodlim)
+set.seed(43500)
+
+# add 0.5 to X2 (with mean 0) to demonstrate the effect of centering
+train_dat <- SimSurv(2000) %>% mutate(X2 = X2 + 0.5)
+test_dat <- SimSurv(20) %>% mutate(X2 = X2 + 0.5)
+test_pred <- test_dat[, 5:6]
+```
+
+We can compare the value of the linear predictor using the true coefficients of 1 with the predictions from the survival package. Plotting them against each other shows the offset of the mean of `X2`.
+
+```{r lp-center}
+fit_survival <- coxph(Surv(time, status) ~ X1 + X2, data = train_dat)
+pred_survival <- predict(fit_survival, newdata = test_pred)
+
+pred <- bind_cols(
+  test_pred %>% mutate(linear_pred = X1 * 1  + X2 * 1),
+  tibble(pred_survival = pred_survival)
+)
+
+mean(train_dat$X2)
+```
+
+```{r lp-center-plot}
+pred %>% 
+  ggplot(aes(linear_pred, pred_survival)) + 
+  geom_point() +
+  geom_abline(slope = 1, intercept = 0)
+```
+
+As part of the modelling workflow, we may want to use these predictions with a performance metric such as concordance. This metric captures the probability that the prediction goes in the same direction as the actual data. While the data captures _survival_ time, the linear predictor for a proportional hazards model is a factor on the _risk_. The predicted survival time is longer if the linear predictor is lower. To make the prediction and the observation go in the same direction, we flip the sign of the prediction. This may be opposite to the sign of predictions obtained using the underlying engine directly.
```

Comment:
We can justify the sign change by saying that our philosophy is that the calculation of performance metrics should be agnostic to ancillary information such as the model type or the training set. Changing the sign helps standardize the model assessment modules.

Reply:
This part is still confusing to me 😔